server/routes/auth/register.route.js

```diff
@@ -16,20 +16,24 @@ router.post("/", (req, res) => {
     return res.status(400).json(errors);
   }
 
-  User.findOne({ email: req.body.email }).then(user => {
+  const name = req.body.name;
+  const email = req.body.email;
+  const password = req.body.password; 
+
+  User.findOne({ email }).then(user => {
     if (user) {
       return res.status(400).json({ email: "Email already exists" });
     } else {
-      const newUser = new User({
-        name: req.body.name,
-        email: req.body.email,
-        password: req.body.password
-      });
+      const newUser = new User({ name, email, password });
       
       newUser.save()
         .then(user => {
-          delete user.password; // remove hashed password from response
-          return res.json(user);
+          const payload = { id: user.id };
+          const token = user.signJWT(payload);
+
+          user.password = null;
```

Comment:
One small comment is to reduce the repeated code for remove the password and possible other things in the future when we want to present the user to the client, we can add another method for the User model to be something like:

user.objectForClient() (I don't like this name at all)

But it'll return an object that is presentable for the client

Reply:
I agree. I'll add that to the icebox. I'll merge this, for now, to test for compatibility with Dhruv's new pull request. And I'll fix this repetition of code by Sunday at the latest